### DISCUSSION TITLE
How to type foreign keys in TypeScript?

### DISCUSSION THREAD

**Discussion by EddieBrunstedt** [supabase/supabase#5773] [Questions]
(I am just starting to learn Supabase and don't know PostgreSQL from before)

I have this key in one of my tables:
```
owner: string;
```
which is a reference to another table called `profiles`. In my app I have some queries that populates this key with the data in the `profiles` table, and some queries that doesn't.

What is the best way to type this in TypeScript? If I make the `owner` key type `owner: ProfileEntity | string` it gets really cumbersome to work with. Maybe I should create two types, one for when I populate the reference and one for when I don't, but that also seems pretty verbose considering there will be *many* different combinations of populating/not populating as the project grows. Am I missing something obvious?

**Comment by madeleineostoja** [comment]
Agree that there should be a higher order generic provided by Supabase to let us more easily type populated foreign table data.

Very simple example from my own project, I have a `photos` table that has a foreign key to a more basic `images` table (which stores a reference to the asset itself). Almost every time I'm fetching a photo I'm fetching the related base image, which means I have this littered everywhere

```js
db.from<definitions['photos'] & { image: definitions['images'] }>('photos')
```

**Comment by pippinmole** [comment]
Bump. I feel like this feature is required if you work with Typescript and foreign keys.

**Comment by joelgfz** [comment]
I am facing the same issue at work, would be great to have a solution other than @madeleineostoja 's. I have been doing that and it feels quite verbose.

**Comment by vsilosangeles** [comment]
Have the same issue, anyone?
